mlir/include/mlir/Dialect/OpenMP/OpenMPOps.td

```diff
@@ -1144,14 +1144,13 @@ def DataBoundsOp : OpenMP_Op<"bounds",
 }
 
 def MapInfoOp : OpenMP_Op<"map_info", [AttrSizedOperandSegments]> {
-  let arguments = (ins OpenMP_PointerLikeType:$var_ptr,
-                       Optional<OpenMP_PointerLikeType>:$var_ptr_ptr,
+  let arguments = (ins AnyTypeOf<[IntLikeType, Index, OpenMP_PointerLikeType]>:$var_ptr,
```

Comment:
I think IntLikeType covers index, it's an alias for: AnyTypeOf<[AnyInteger, Index]>

Reply:
The design seems wrong here. Why do you need to allow for an integer here? The argument name suggests a `ptr` right now.

Reply:
In this case I believe it's because index values are sometimes used inside of the omp.target region when generated outside of the omp.target region and we capture them implicitly in this patch as there may be a need to pass them to the device, I imagine this could perhaps be restricted to just Index, but I agree OpenMP_PointerLikeType would be nice to have on its own. @TIFitis can likely explain it better than I can! 

I think Kiran suggested a name change if we continue down this route, or an alternative field.

Reply:
I agree with Valentin about this particular change being wrong.
To me, it seems OK to have a different constraint for variables mapped by value - with a different operand. And then allowing only `varPtr` or `var`.

Reply:
The latest revision resolves this.

Reply:
Wouldn't the `omp.bounds` operation for `a` have  `%4` for `n`?

One way to extend `omp.map_info` would be to add another optional argument ($val) to `MapInfo` and make $var_ptr optional. Modify the verifier to ensure that one of $val and $var_ptr is always provided but both are not provided.